notebooks/nb_utils/lesson_preprocessor.py

```diff
@@ -48,14 +48,28 @@ def preprocess(self, nb, resources):
             c = c and macroer.process_cell(c)
             if c is not None:
                 new_cells.append(c)
+
+        # Add header and footer for self-passed courses, but not for daily-email notebooks (where it'd be confusing)
+        if not self.cfg.get('daily'):
```

Comment:
Replacing `daily` with an enum would be really nice if possible.  (Not that you have to do it in this PR, just a note for the future.)

Reply:
daily is set in a metadata file like 
https://github.com/Kaggle/learntools/blob/master/notebooks/machine_learning/daily.yaml

Is the idea is to either:
1. Convert the dict created when a parse that yaml file into a class, where I have a daily property that could be an enum (though for that matter it could just be a bool).
2. Go through the dict and convert daily and similar values into enums?

Reply:
ah...  Interesting problem.  

For historical reasons, a lot of the yaml files currently don't have 'daily' in them. That implicitly means 'daily' is false.

We could go back and add `'daily': False` to those old Yamls, and then assert that 'daily' is in there (e.g. retrieve it as `self.cfg['daily']`) . 

Short of that, it seems hard to prevent the type of problem you bring up up.